nix/default.nix

```diff
@@ -46,129 +45,130 @@
-    ];
+      COMMITS = revCount;
+      DATE = date;
+      DIRTY = lib.optionalString (commit == "") "dirty";
+      HASH = commit;
 
-    outputs = [
-      "out"
-      "man"
-      "dev"
-    ];
+      depsBuildBuild = [
+        pkg-config
+      ];
 
-    buildInputs = lib.concatLists [
-      [
-        aquamarine
-        cairo
-        # expat
-        # fribidi
-        git
-        hyprcursor
-        hyprlang
-        hyprutils
-        # libdatrie
-        libdrm
-        libGL
-        libinput
-        # libselinux
-        # libsepol
-        # libthai
-        libuuid
-        libxkbcommon
-        mesa
-        pango
-        pciutils
-        # pcre2
-        tomlplusplus
-        wayland
-        wayland-protocols
-        xorg.libXcursor
-      ]
-      (lib.optionals stdenv.hostPlatform.isMusl [libexecinfo])
-      (lib.optionals enableXWayland [
-        xorg.libxcb
-        xorg.libXdmcp
-        xorg.xcbutilerrors
-        xorg.xcbutilrenderutil
-        xorg.xcbutilwm
-        xwayland
-      ])
-      (lib.optionals withSystemd [systemd])
-    ];
+      nativeBuildInputs = [
+        cmake
+        hyprwayland-scanner
+        jq
+        makeWrapper
+        pkg-config
+        python3 # for udis86
+        wayland-scanner
+      ];
 
-    mesonBuildType =
-      if debug
-      then "debug"
-      else "release";
+      outputs = [
+        "out"
+        "man"
+        "dev"
+      ];
 
-    # we want as much debug info as possible
-    dontStrip = debug;
+      buildInputs = lib.concatLists [
+        [
+          aquamarine
+          cairo
+          git
+          hyprcursor
+          hyprlang
+          hyprutils
+          libdrm
+          libGL
+          libinput
+          libuuid
+          libxkbcommon
+          mesa
+          pango
+          pciutils
+          tomlplusplus
+          wayland
+          wayland-protocols
+          xorg.libXcursor
+        ]
+        (lib.optionals stdenv.hostPlatform.isMusl [libexecinfo])
+        (lib.optionals enableXWayland [
+          xorg.libxcb
+          xorg.libXdmcp
+          xorg.xcbutilerrors
+          xorg.xcbutilrenderutil
+          xorg.xcbutilwm
+          xwayland
+        ])
+        (lib.optionals withSystemd [systemd])
+      ];
 
-    mesonFlags = [
-      (lib.mesonEnable "xwayland" enableXWayland)
-      (lib.mesonEnable "legacy_renderer" legacyRenderer)
-      (lib.mesonEnable "systemd" withSystemd)
-    ];
+      cmakeBuildType =
+        if debug
+        then "Debug"
+        else "RelWithDebInfo";
 
-    postInstall = ''
-      ${lib.optionalString wrapRuntimeDeps ''
-        wrapProgram $out/bin/Hyprland \
-          --suffix PATH : ${lib.makeBinPath [
-          binutils
-          pciutils
-          pkgconf
-        ]}
-      ''}
-    '';
+      cmakeFlags = [
+        # precompiled headers are useless in Nix
+        "-DCMAKE_DISABLE_PRECOMPILE_HEADERS=ON"
+        (lib.cmakeBool "NO_XWAYLAND" (!enableXWayland))
```

Comment:
There is a nice little interface here to simplify it to a set.

https://github.com/NixOS/nixpkgs/blob/4f807e8940284ad7925ebd0a0993d2a1791acb2f/pkgs/by-name/uw/uwsm/package.nix#L57-L63

Reply:
Why did I not think of this earlier...